eva-accession-import/src/main/java/uk/ac/ebi/eva/accession/dbsnp/contig/ContigMapping.java

```diff
@@ -57,68 +49,45 @@ public ContigMapping(AssemblyReportReader assemblyReportReader) throws Exception
-        if (assignedMoleculeDuplicates.contains(contigSynonym.getAssignedMolecule())) {
-            contigSynonym.setAssignedMolecule(null);
-        }
     }
 
     private void populateMaps(AssemblyReportReader assemblyReportReader) throws Exception {
         ContigSynonyms contigSynonyms;
         while ((contigSynonyms = assemblyReportReader.read()) != null) {
             fillContigConventionMaps(contigSynonyms);
         }
-        if (!assignedMoleculeDuplicates.isEmpty()) {
-            removeDuplicatedAssignedMolecule();
-        }
     }
 
     /**
      * Adds an entry to every map, where the key is a name, and the value is the row where it appears.
      *
      * Take into account:
-     * - sequenceName and UCSC columns may have prefixes that we have to remove.
      * - UCSC and assignedMolecule columns may appear as "na" (not available).
-     * - assignedMolecule values may not be unique across rows. If that's the case, don't include assignedMolecule in
-     * the maps.
+     * - assignedMolecule values may not be unique across rows. Keep only those that have "assembled-molecule" in the
+     * Sequence-Role column.
      */
     private void fillContigConventionMaps(ContigSynonyms contigSynonyms) {
         normalizeNames(contigSynonyms);
 
         sequenceNameToSynonyms.put(contigSynonyms.getSequenceName(), contigSynonyms);
-        genBankToSynonyms.put(contigSynonyms.getGenBank(), contigSynonyms);
-        refSeqToSynonyms.put(contigSynonyms.getRefSeq(), contigSynonyms);
-
+        if (contigSynonyms.getAssignedMolecule() != null) {
+            assignedMoleculeToSynonyms.put(contigSynonyms.getAssignedMolecule(), contigSynonyms);
+        }
+        if (contigSynonyms.getGenBank() != null) {
+            genBankToSynonyms.put(contigSynonyms.getGenBank(), contigSynonyms);
+        }
+        if (contigSynonyms.getRefSeq() != null) {
+            refSeqToSynonyms.put(contigSynonyms.getRefSeq(), contigSynonyms);
+        }
         if (contigSynonyms.getUcsc() != null) {
             ucscToSynonyms.put(contigSynonyms.getUcsc(), contigSynonyms);
         }
 
-        if (contigSynonyms.getAssignedMolecule() != null) {
-            ContigSynonyms previousValue = assignedMoleculeToSynonyms.putIfAbsent(contigSynonyms.getAssignedMolecule(),
-                                                                                  contigSynonyms);
-            if (previousValue != null) {
-                assignedMoleculeDuplicates.add(contigSynonyms.getAssignedMolecule());
-            }
-        }
     }
 
     private void normalizeNames(ContigSynonyms contigSynonyms) {
-        contigSynonyms.setSequenceName(removePrefix(contigSynonyms.getSequenceName()));
-        if (contigSynonyms.getAssignedMolecule().equals(NOT_AVAILABLE)) {
+        if (contigSynonyms.getAssignedMolecule().equals(NOT_AVAILABLE)
+                || !ASSEMBLED_MOLECULE.equals(contigSynonyms.getSequenceRole())) {
```

Comment:
`NOT_AVAILABLE.equals(contigSynonyms.getAssignedMolecule())` for consistent ordering in both conditions (and avoidance of NPE).